pl_examples/hydra_examples/conf/optimizer.py

```diff
@@ -0,0 +1,98 @@
+from dataclasses import dataclass
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+@dataclass
+class AdamConf:
+    betas: tuple = (0.9, 0.999)
+    lr: float = 1e-3
+    eps: float = 1e-8
+    weight_decay: float = 0
+    amsgrad: bool = False
+
+
+@dataclass
+class AdamaxConf:
+    betas: tuple = (0.9, 0.999)
+    lr: float = 1e-3
+    eps: float = 1e-8
+    weight_decay: float = 0
+
+
+@dataclass
+class ASGDConf:
+    alpha: float = 0.75
+    lr: float = 1e-3
+    lambd: float = 1e-4
+    t0: int = 1e6
+    weight_decay: float = 0
+
+
+@dataclass
+class LBFGSConf:
+    lr: float = 1
+    max_iter: int = 20
+    max_eval: int = 25
+    tolerance_grad: float = 1e-5
+    tolerance_change: float = 1e-9
+    history_size: int = 100
+    line_search_fn: str = None
+
+
+@dataclass
+class RMSpropConf:
+    lr: float = 1e-2
+    momentum: float = 0
+    alpha: float = 0.99
+    eps: float = 1e-8
+    centered: bool = True
+    weight_decay: float = 0
+
+
+@dataclass
+class RpropConf:
+    lr: float = 1e-2
+    momentum: tuple = (0.5, 1.2)
+    step_sizes: tuple = (1e-6, 50)
+
+
+@dataclass
+class SGDConf:
+    lr: float = 1e-2
+    momentum: float = 0
+    weight_decay: float = 0
+    dampening: float = 0
+    nesterov: bool = False
+
+
+ConfigStore.instance().store(
```

Comment:
Imo it'd make more sense to put the .store(...) right under the param so you can quickly see which config relates to which cls

Reply:
Fixed!